L1Trigger/L1TTrackMatch/plugins/L1TkElectronTrackProducer.cc

```diff
@@ -0,0 +1,340 @@
+
+      if (IsoCut <= 0) {
+        // write the L1TkEm particle to the collection,
+        // irrespective of its relative isolation
+        result->push_back(trkEm);
+      } else {
+        // the object is written to the collection only
+        // if it passes the isolation cut
+        if (trkisol <= IsoCut)
+          result->push_back(trkEm);
+      }
+    }
+
+  }  // end loop over EGamma objects
+
+  iEvent.put(std::move(result), label);
+}
+
+// ------------ method called once each job just before starting event loop  ------------
+void L1TkElectronTrackProducer::beginJob() {}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void L1TkElectronTrackProducer::endJob() {}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+L1TkElectronTrackProducer::beginRun(edm::Run& iRun, edm::EventSetup const& iSetup)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+L1TkElectronTrackProducer::endRun(edm::Run&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+L1TkElectronTrackProducer::beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+L1TkElectronTrackProducer::endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void L1TkElectronTrackProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
+// method to calculate isolation
+float L1TkElectronTrackProducer::isolation(const edm::Handle<L1TTTrackCollectionType>& trkHandle, int match_index) {
+  edm::Ptr<L1TTTrackType> matchedTrkPtr(trkHandle, match_index);
+  L1TTTrackCollectionType::const_iterator trackIter;
+
+  float sumPt = 0.0;
+  int itr = 0;
+  for (trackIter = trkHandle->begin(); trackIter != trkHandle->end(); ++trackIter) {
+    if (itr++ != match_index) {
+      if (trackIter->chi2() > maxChi2IsoTracks || trackIter->momentum().perp() <= PTMINTRA ||
+          trackIter->getStubRefs().size() < minNStubsIsoTracks) {
+        continue;
+      }
+
+      float dZ = fabs(trackIter->POCA().z() - matchedTrkPtr->POCA().z());
+
+      float phi1 = trackIter->momentum().phi();
+      float phi2 = matchedTrkPtr->momentum().phi();
+      float dPhi = reco::deltaPhi(phi1, phi2);
+      float dEta = (trackIter->momentum().eta() - matchedTrkPtr->momentum().eta());
+      float dR = sqrt(dPhi * dPhi + dEta * dEta);
```

Comment:
consider just using dR^2 to avoid sqrt (DRmin^2 and DRmax^2 can be stored once at the beginning)

Reply:
done